front/src/common/SubCategoryContext.tsx

```diff
@@ -0,0 +1,51 @@
+import { createContext, useContext, useEffect, useMemo } from 'react';
+
+import { useTranslation } from 'react-i18next';
+
+import { osrdEditoastApi } from 'common/api/osrdEditoastApi';
+import type { SubCategory } from 'common/api/osrdEditoastApi';
+import { setFailure } from 'reducers/main';
+import { useAppDispatch } from 'store';
+
+type SubCategoryContextValue = {
+  subCategories: SubCategory[];
+};
+
+const SubCategoryContext = createContext<SubCategoryContextValue>({ subCategories: [] });
```

Comment:
nit: you can directly return the subcategories, no need to have an object around ;) 
```suggestion
const SubCategoryContext = createContext<SubCategoryContextValue>([]);
```

Reply:
Done